leaflet-geoman.d.ts

```diff
@@ -1,340 +1,785 @@
-            | 'pm:unsnap'
-            | 'pm:intersect'
-            | 'pm:centerplaced';
-
-        export type GEOMAN_LAYER_DRAG_MODE_EVENTS =
-            | 'pm:dragstart'
-            | 'pm:drag'
-            | 'pm:dragend';
-
-        export type GEOMAN_LAYER_CUT_MODE_EVENTS = 'pm:cut' | 'pm:edit';
-
-        export interface GeomanHelpers {
-            getShapes(): string[];
-            getActiveShape(): string;
+            | 'CircleMarker'
+            | 'ImageOverlay';
+
+        /** supported shape types. 'ImageOverlay' is in Edit Mode only. */
+        type SUPPORTED_SHAPE_TYPES =
+            | L.Marker
+            | L.Circle
+            | L.Polyline
+            | L.Rectangle
+            | L.Polygon
+            | L.CircleMarker
+            | L.ImageOverlay;
+
+
+        /**
+        * Changes default registration of leaflet-geoman on leaflet layers.
+        *
+        * @param optIn - if true, a layers pmIgnore property has to be set to false to get initiated.
+        */
+        function setOptIn(optIn: boolean): void;
+
+        /**
+         * Enable leaflet-geoman on an ignored layer.
+         *
+         * @param layer - re-reads layer.options.pmIgnore to initialize leaflet-geoman.
+         */
+        function reInitLayer(layer: L.Layer): void;
+
+
+        /**
+         * PM map interface.
+         */
+        interface PMMap extends PMDrawMap, PMEditMap, PMDragMap, PMRemoveMap, PMCutMap {
+
+            Toolbar: PMMapToolbar;
+
+            /** Adds the Toolbar to the map. */
+            addControls(options?: ToolbarOptions): void;
+
+            /** Toggle the visiblity of the Toolbar. */
+            removeControls(): void;
+
+            /** Returns true if the Toolbar is visible on the map. */
+            controlsVisible(): boolean;
+
+            /** Toggle the visiblity of the Toolbar. */
+            toggleControls(): void;
+
+            setLang(
+                lang:
+                    | 'da'
+                    | 'de'
+                    | 'el'
+                    | 'en'
+                    | 'es'
+                    | 'fa'
+                    | 'fr'
+                    | 'hu'
+                    | 'id'
+                    | 'it'
+                    | 'nl'
+                    | 'no'
+                    | 'pl'
+                    | 'pt_br'
+                    | 'ro'
+                    | 'ru'
+                    | 'sv'
+                    | 'tr'
+                    | 'ua'
+                    | 'zh'
+                    | 'zh_tw',
+                customTranslations?: Translations,
+                fallbackLanguage?: string
+            ): void;
+
+            /** Set globalOptions and apply them. */
+            setGlobalOptions(options: GlobalOptions): void;
+
+            /** Apply the current globalOptions to all existing layers. */
+            applyGlobalOptions(): void;
+
+            /** Returns the globalOptions. */
+            getGlobalOptions(): GlobalOptions;
```

Comment:
are these the only ones that should be moved from PMDrawMap?

Reply:
Yes